### ISSUE TITLE
Event will not be registered in Google if `label` and `value` is not specified

### ISSUE THREAD

**Issue by andrewspy** [wasc-io/iron-ga#1]
Below is a sample code, which should register an event in GA.

```js
this.dispatchEvent(new CustomEvent('track-event', {
  composed: true,
  bubbles: true,
  detail: {
    category: 'messages',
    action: 'send_text_message',
    // label: 'group', // Optional
    // value: 1 // Optional
  }
}));
```

**Comment by NicolaiSchmid** [comment]
Was exactly is your issue?
Does your snippet above not produce a trackevent?

**Comment by andrewspy** [comment]
It does produce a `track-event` with "Data reported to Google Analytics" being shown in the console (BTW, it's better to only show that behind a `log` flag). However, on Google side, the event was not registered/pick-up for whatever reason.

When I tried to include both `label` and `value` with the given value (i.e. by un-commenting the lines), it can register in Google with no issue.

**Comment by NicolaiSchmid** [comment]
This problem seems weird, since the [`hitCallback`](https://github.com/wasc-io/iron-ga/blob/master/iron-ga.html#L116)` Function is executed by the Google Analytics script itself.
Maybe there's an error reporting the event to the Google Servers.
Have you checked the network section in the DevTools to see, if any data is sent to Google?

**Comment by andrewspy** [comment]
I think I have discovered what's the issue. On https://github.com/wasc-io/iron-ga/blob/master/iron-ga.html#L114, you are not suppose to send `eventLabel` and `eventValue` if it is not defined, which you did by initializing both to an empty string `''`.

According to [Google Analytics Event Tracking Doc](https://developers.google.com/analytics/devguides/collection/analyticsjs/events), `eventValue` has to be an integer, which if you set it as `''`, it will be send as `''` from the DevTools, and will not be registered on Google's end (although it did show up on network section of Devtools).

So to resolve the issue, I think the best way is to follow as per Google Analytics doc by not sending `eventLabel` and `eventValue` if it's not defined.

**Comment by NicolaiSchmid** [comment]
Okay, I will fix the issue later

**Comment by NicolaiSchmid** [comment]
I've created a new commit and a new tag v2.0.9 which should fix the issue 5c61c1e3c6dc9650e739a4725bb4569f3e429014
Can you update the element and test it in your setup?

**Comment by andrewspy** [comment]
I have tested v2.0.9 and it is working as expected now.

As for the `window.debug`, I suggest to use a `log` boolean attribute as a handy way to enable logging in markup, which I shall discuss in a separate issue.

Thanks for fixing it, and I shall close this issue.
